apps/vessel-history/feature/download-activity/download-activity.tsx

```diff
@@ -0,0 +1,151 @@
+import { Fragment, useCallback, useState } from 'react'
+import cx from 'classnames'
+import { useTranslation } from 'react-i18next'
+import { ButtonType, IconButton, Modal, Spinner } from '@globalfishingwatch/ui-components'
+import styles from './download-activity.module.css'
+import useDownloadActivity from './download-activity.hook'
+
+export interface DownloadActivityProps {
+  filtersApplied: boolean
+  onDownloadAllActivityCsv?: () => void
+  onDownloadFilteredActivityCsv?: () => void
+  onReadmeClick?: () => void
+  type?: ButtonType
+}
+
+export function DownloadActivity(props: DownloadActivityProps) {
+  const { t } = useTranslation()
+  const { filtersApplied, onDownloadAllActivityCsv, onDownloadFilteredActivityCsv, onReadmeClick } =
+    props
+  const [showDownloadPopup, setShowDownloadPopup] = useState(false)
+  const { downloadAllEvents, downloadFilteredEvents, downloadingStatus, noEvents, readmeUrl } =
+    useDownloadActivity()
+
+  const handleCloseDownloadPopup = useCallback(() => {
+    setShowDownloadPopup(false)
+  }, [])
+
+  const handleDownloadAllActivityCsv = useCallback(() => {
+    downloadAllEvents()
+    if (onDownloadAllActivityCsv !== undefined) {
+      onDownloadAllActivityCsv()
+    }
+    handleCloseDownloadPopup()
+  }, [downloadAllEvents, handleCloseDownloadPopup, onDownloadAllActivityCsv])
+
+  const handleDownloadFilteredActivityCsv = useCallback(() => {
+    downloadFilteredEvents()
+    if (onDownloadFilteredActivityCsv !== undefined) {
+      onDownloadFilteredActivityCsv()
+    }
+    handleCloseDownloadPopup()
+  }, [downloadFilteredEvents, handleCloseDownloadPopup, onDownloadFilteredActivityCsv])
+
+  const handleReadmeClick = useCallback(() => {
+    if (onReadmeClick !== undefined) {
+      onReadmeClick()
+    }
+    handleCloseDownloadPopup()
+  }, [handleCloseDownloadPopup, onReadmeClick])
+
+  return (
+    <Fragment>
+      {!downloadingStatus && (
+        <Fragment>
+          <IconButton
+            type={props?.type === 'default' ? 'map-tool' : 'solid'}
+            icon={showDownloadPopup ? 'close' : 'download'}
+            size="medium"
+            onClick={() => setShowDownloadPopup(!showDownloadPopup)}
+          />
+
+          {showDownloadPopup && (
+            <Modal
+              appSelector="__next"
+              className={styles.modalContentWrapper}
+              closeButtonClassName={styles.modalCloseButton}
+              contentClassName={styles.modalContainer}
+              header={false}
+              isOpen={showDownloadPopup}
+              onClose={handleCloseDownloadPopup}
+              overlayClassName={styles.modalOverlay}
+              portalClassName={styles.modalPortal}
+              shouldCloseOnEsc={true}
+            >
+              <ul className={styles.items}>
+                {!noEvents && (
```

Comment:
I think a validation like `{hasEvents && ...` is easier to understand than double negations, WDYT?